server/src/main/java/org/elasticsearch/action/ingest/SimulatePipelineTransportAction.java

```diff
@@ -35,30 +61,79 @@ public SimulatePipelineTransportAction(
         super(SimulatePipelineAction.NAME, transportService, actionFilters, SimulatePipelineRequest::new);
         this.ingestService = ingestService;
         this.executionService = new SimulateExecutionService(threadPool);
+        this.transportService = transportService;
+        this.ingestNodeTransportActionTimeout = INGEST_NODE_TRANSPORT_ACTION_TIMEOUT.get(ingestService.getClusterService().getSettings());
+        ingestService.getClusterService()
+            .getClusterSettings()
+            .addSettingsUpdateConsumer(
+                INGEST_NODE_TRANSPORT_ACTION_TIMEOUT,
+                newTimeout -> this.ingestNodeTransportActionTimeout = newTimeout
+            );
     }
 
     @Override
     protected void doExecute(Task task, SimulatePipelineRequest request, ActionListener<SimulatePipelineResponse> listener) {
         final Map<String, Object> source = XContentHelper.convertToMap(request.getSource(), false, request.getXContentType()).v2();
-
-        final SimulatePipelineRequest.Parsed simulateRequest;
+        DiscoveryNodes discoveryNodes = ingestService.getClusterService().state().nodes();
+        Map<String, DiscoveryNode> ingestNodes = discoveryNodes.getIngestNodes();
+        if (ingestNodes.isEmpty()) {
+            /*
+             * Some resources used by pipelines, such as the geoip database, only exist on ingest nodes. Since we only run pipelines on
+             * nodes with the ingest role, we ought to only simulate a pipeline on nodes with the ingest role.
+             */
+            listener.onFailure(
+                new ElasticsearchException(
+                    "Cannot run a pipeline simulation because there are no nodes in the cluster with the ingest role"
+                )
+            );
```

Comment:
I think this needs a `return;` statement so we skip the rest of the processing if there isn't a node I think?

Reply:
I actually removed this just now because we have a test that makes sure we can run this even if there is no ingest node. I changed it to log an info message in that case and to make a best effort to run the pipeline simulation. I'd like to fail the simulation, but I think that this probably makes more sense for backward compatibility. How about you?

Reply:
I'm good with that. Done.